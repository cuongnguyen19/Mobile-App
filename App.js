import React from "react";

import { NavigationContainer } from "@react-navigation/native";
import InitialNavigator from "./app/navigation/InitialNavigator";

export default function App() {
  return (
    <NavigationContainer>
      <InitialNavigator/>
    </NavigationContainer>


  );
}
