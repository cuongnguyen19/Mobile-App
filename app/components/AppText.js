//Import all necessary react components 
import React from "react";
import { Text, StyleSheet, Platform } from "react-native";

//File/Function for all the texts in the app
function AppText({ children, style }) {
  return <Text style={[styles.text, style]}> {children}</Text>;
}

//Style for the text in the app
const styles = StyleSheet.create({
  text: {
    fontSize: 20,
    fontFamily: Platform.OS == "android" ? "monospace" : "Avenir-Roman",
  },
});

export default AppText;
