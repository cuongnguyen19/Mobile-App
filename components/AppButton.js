//Import all necessary react components 
import React from "react";
import { Text, TouchableOpacity, View, StyleSheet } from "react-native";

//Import all necessary files for use in this function
import AppColors from "../config/AppColors";

//File/Function for the buttons in the app with the specified styles, title, and passed onpress function to handle
function AppButton({ title, style, textStyle, color = "primaryColor", onPress, testid }) {
  return (
    <TouchableOpacity onPress={onPress} testID = {testid}>
      <View style={[styles.button, style, { backgroundColor: AppColors[color] }]}>
        <Text style={[styles.text, textStyle]}> {title}</Text>
      </View>
    </TouchableOpacity>
  );
}

//Styles (Layout) for the components in the app buttons
const styles = StyleSheet.create({
  button: {
    width: "100%",
    borderRadius: 20,
    padding: 15,
    justifyContent: "center",
    alignItems: "center",
    bottom: 50,
  },
  text: {
    color: AppColors.black,
    fontSize: 15,
    textTransform: "uppercase",
    fontWeight: "bold",
  },
});

export default AppButton;
