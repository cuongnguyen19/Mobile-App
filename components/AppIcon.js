//Import all necessary react components 
import React from 'react';
import { View } from 'react-native';
//Import the icon library for use within the app
import {MaterialCommunityIcons} from '@expo/vector-icons';

//File/Function for the icons in the app with the specified name, size and color.
function AppIcon({name, size=40, iconColor="black", backgroundColor}) {
    return (
        <View style={{width: size, height: size, backgroundColor, borderRadius: size/2, alignItems:"center", justifyContent:"center"}}>
            <MaterialCommunityIcons name={name} size={size*0.6} color={iconColor}/>
        </View>
    );
}

export default AppIcon;