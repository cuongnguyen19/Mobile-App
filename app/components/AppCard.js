//Import all necessary react components 
import React from 'react';
import { View, Image, StyleSheet } from 'react-native';

//Import all necessary files for use in this function
import AppColors from '../config/AppColors';
import AppText from './AppText';

/*File/Function for the cards of places in the app with the specified title (place name), subtitle (address),
category and image with all the specified styles */
function AppCard({category, title, subtitle, image, style}) {
    return (
        <View style={[styles.container, style]}>
            {isFinite(image)? <Image source={image} style={styles.image}/> : <Image source={{uri: image}} style={styles.image}/> }
            <AppText style={styles.title}> {title} - {category} </AppText>
            <AppText style={styles.subtitle}> {subtitle} </AppText>          
        </View>
    );
}

//Styles (Layout) for the title, subtitle, image
const styles = StyleSheet.create({
    container:{
        backgroundColor:AppColors.otherColorLite,
        borderRadius:30,
        overflow:"hidden",
        marginTop: 25,
        alignItems:"center",
    }, 
    image:{
        height: 200,
        width: "100%",
    },
    title:{
        fontWeight:"bold",
        fontSize:25,
    },
    subtitle:{
        fontSize:18,
    }
})

export default AppCard;