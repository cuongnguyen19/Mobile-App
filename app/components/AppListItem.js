//Import all necessary react components 
import React from 'react';
import { View, StyleSheet, Image, TouchableHighlight } from 'react-native';
import Swipeable from 'react-native-gesture-handler/Swipeable';

//Import all necessary files for use in this function
import AppColors from '../config/AppColors';
import AppText from './AppText';

/*File/Function for the User Experience of the app.
With this function, the users can either click on an item (a card) 
to see more information about it, or swipe left to delete it*/
function AppListItem({image, title, subtitle, AppComponent, onPress, onSwipeLeft, style, status = false}) {
    return (
        //Function for swiping and touching
        <Swipeable renderRightActions={onSwipeLeft}>
            <TouchableHighlight onPress={onPress} underlayColor={AppColors.otherColorLite}>
                <View style = {style}>
                    {AppComponent}
                    {image && <Image source={image} style={styles.image}/>}
                    {status && <View style={styles.textContainer}>
                        <AppText style={styles.title}> {title} </AppText>
                        {subtitle && <AppText style={styles.subtitle}> {subtitle} </AppText>}
                    </View>}
                </View>
            </TouchableHighlight>
        </Swipeable>
    );
}

//Styles (Layout) for all the texts and image in the app
const styles = StyleSheet.create({
    container:{
        flexDirection:"row",
        padding:10,
    },
    image:{
        height: 75,
        width: 75,
        borderRadius: 37,
        marginLeft: 10,
    },
    textContainer:{
        flexDirection:"column",
        marginLeft: 20,
    },
    title:{
        fontWeight:"bold",
        marginVertical: 5,
    },
    subtitle:{
        fontSize:15,
        marginLeft: 5,
    }
    
})

export default AppListItem;