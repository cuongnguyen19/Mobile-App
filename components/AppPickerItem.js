//Import all necessary react components 
import React from 'react';
import { TouchableOpacity, StyleSheet, Image } from 'react-native';

//Import all necessary files for use within this function
import AppIcon from './AppIcon';
import AppText from './AppText';

/*File/Function for each of the 'clickable' item in the dropdown list (modal) 
with its own label, icon, image, color and onpress function to handle*/
function AppPickerItem({onPress, label, icon, image, backgroundColor}) {
    return (
        <TouchableOpacity onPress={onPress} style={styles.container}>
            {image && <Image
                    source={image}
                    style = {styles.image}
                />}
            {icon && <AppIcon name={icon} iconColor="white" backgroundColor={backgroundColor}/>}
            <AppText> {label} </AppText>
        </TouchableOpacity>
    );
}

//Styles (Layout) for the view and image
const styles = StyleSheet.create({
    container:{
        paddingHorizontal:10,
        paddingVertical: 10,
        width:"33%",
        alignItems:"center",
    },
    image:{
        height: 50, 
        width: 50,
    }
})

export default AppPickerItem;