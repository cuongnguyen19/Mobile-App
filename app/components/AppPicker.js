//Import all necessary react components 
import React, {useState} from 'react';
import { View, StyleSheet, Modal, Button, TouchableWithoutFeedback, FlatList } from 'react-native';
//Import the icon library for use within the app
import {MaterialCommunityIcons} from '@expo/vector-icons';

//Import all necessary files for use within this function
import AppText from './AppText';
import AppScreen from './AppScreen';
import AppPickerItem from './AppPickerItem';

/*File/Function for the list of items that the users can pick from 
when they click on the modal view (the search function) in the app.
It is just similar to the dropdown list, which displays items that users can choose from*/
function AppPicker({data, icon, placeholder, numColumns, selectedItem, onSelectItem, onPress}) {
    const [modalVisible, setModalVisible] = useState(false);
    return (
        //Function for users to click on to see the dropdown list
        <>
            <TouchableWithoutFeedback onPress={() => setModalVisible(true)}>
                <View style={styles.container}>
                    {icon && <MaterialCommunityIcons name={icon} size={24}/>}
                    <AppText style={styles.text}> {selectedItem? selectedItem.label : placeholder} </AppText>
                    <MaterialCommunityIcons name="chevron-down" size={24}/>
                </View>
            </TouchableWithoutFeedback>

            <Modal visible={modalVisible} animationType="slide">
                <AppScreen>
                    <Button title="Close" onPress={() => setModalVisible(false)}/>
                    <FlatList
                    //Flat list to display a list of items in the modal (dropdown list)
                        numColumns={numColumns}
                        data={data}
                        keyExtractor={item => item.value.toString()}
                        renderItem = {({item}) => 
                        /*Each item in the modal (dropdown list) with its own 
                        label, image or icon, and color that users can choose to click*/
                        <AppPickerItem
                            onPress={()=> {
                                {onPress && onPress()};
                                setModalVisible(false);
                                onSelectItem(item);
                                    }
                                }
                            image = {item.image}
                            label={item.label}
                            icon={item.icon}
                            backgroundColor={item.backgroundColor}
                            />}
                    />
                </AppScreen> 
            </Modal>
        </>
    );
}

//Styles (Layout) for the texts and view in the app
const styles = StyleSheet.create({
    container:{
        backgroundColor:"#e0e0eb",
        flexDirection: 'row',
        borderRadius: 25, 
        padding: 10,
        marginVertical: 20,
        width:'100%',
    },
    text:{
        flex:1,
    },
})
export default AppPicker;