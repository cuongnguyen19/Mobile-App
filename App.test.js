// Import all necessary react and testing components
import React from "react";
import renderer from "react-test-renderer";
import {render, fireEvent, act} from "@testing-library/react-native";
import {configure, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

// Import all necessary files and custom components
import AppButton from "./app/components/AppButton";
import AppCard from "./app/components/AppCard";
import AppIcon from "./app/components/AppIcon";
import AppListItem from "./app/components/AppListItem";
import AppPicker from "./app/components/AppPicker";
import AppPickerItem from "./app/components/AppPickerItem";
import AppScreen from "./app/components/AppScreen";
import AppText from "./app/components/AppText";
import AppTextInput from "./app/components/AppTextInput";
import SigninScreen from "./app/screens/SigninScreen";
import App from "./App";

// Configure the adapter and mock the animation for testing
configure({adapter: new Adapter()});
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

// Test the 'changed' status of every component
describe("<AppButton />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppCard />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppCard />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppIcon />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppIcon />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppListItem />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppListItem />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppPicker />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppPicker />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppPickerItem />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppPickerItem />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppScreen />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppText />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppText />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
describe("<AppTextInput />", () => {
    test('renders unchanged', () => {
        const tree = renderer.create(<AppTextInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// Test that the app runs without crash or errors
describe("<App />", () => {
    test('renders without crash', () => {
        const snap = shallow(<App />);
        expect(snap).toMatchSnapshot();
    });
});

// Test the signin function with valid or right input credentials
describe("SigninScreen />", () => {
    test('input fields render correctly and navigation works with valid or right inputs', async() => {
        const onSubmit = jest.fn();
        const navigate = jest.fn();
        const {getByPlaceholderText, getByTestId} = render(<SigninScreen onClick = {onSubmit} navigation = {{navigate}} />);
        const emailInput = getByPlaceholderText('Email Address');
        const passwordInput = getByPlaceholderText('Password');
        const signinButton = getByTestId("signin");

        await act(async() => {
        fireEvent.changeText(emailInput, 'm@gmail.com');
        });
        expect(emailInput.props.value).toBe('m@gmail.com');

        await act(async() => {
        fireEvent.changeText(passwordInput, '1234');
        });
        expect(passwordInput.props.value).toBe('1234');

        await act(async() => {
            fireEvent(signinButton, 'press');
        });

        expect(onSubmit).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledTimes(1);

    });

    // Test the signin function with invalid or wrong input credentials
    test('input fields render correctly and navigation does not work with invalid or wrong inputs', async() => {
        const onSubmit = jest.fn();
        const navigate = jest.fn();
        const {getByPlaceholderText, getByTestId} = render(<SigninScreen onClick = {onSubmit} navigation = {{navigate}} />);
        const emailInput = getByPlaceholderText('Email Address');
        const passwordInput = getByPlaceholderText('Password');
        const signinButton = getByTestId("signin");

        await act(async() => {
        fireEvent.changeText(emailInput, 'm@gmail');
        });
        expect(emailInput.props.value).toBe('m@gmail');
        
        await act(async() => {
        fireEvent.changeText(passwordInput, '1234');
        });
        expect(passwordInput.props.value).toBe('1234');

        await act(async() => {
            fireEvent(signinButton, 'press');
        });

        expect(onSubmit).toHaveBeenCalledTimes(0);
        expect(navigate).toHaveBeenCalledTimes(0);
    });
});


