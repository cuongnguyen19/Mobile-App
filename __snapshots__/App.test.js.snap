// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crash 1`] = `ShallowWrapper {}`;

exports[`<AppButton /> renders unchanged 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "bottom": 50,
          "justifyContent": "center",
          "padding": 15,
          "width": "100%",
        },
        undefined,
        Object {
          "backgroundColor": "#4e5c50",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 15,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          undefined,
        ]
      }
    >
       
    </Text>
  </View>
</View>
`;

exports[`<AppCard /> renders unchanged 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2eadf",
        "borderRadius": 30,
        "marginTop": 25,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": 200,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
        },
      ]
    }
  >
     
     
     - 
     
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        Object {
          "fontSize": 18,
        },
      ]
    }
  >
     
     
     
  </Text>
</View>
`;

exports[`<AppIcon /> renders unchanged 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <Text />
</View>
`;

exports[`<AppListItem /> renders unchanged 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`<AppPicker /> renders unchanged 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0eb",
        "borderRadius": 25,
        "flexDirection": "row",
        "marginVertical": 20,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Avenir-Roman",
            "fontSize": 20,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
       
       
       
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó°…€
    </Text>
  </View>,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </RCTSafeAreaView>
  </Modal>,
]
`;

exports[`<AppPickerItem /> renders unchanged 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
      "width": "33%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Avenir-Roman",
          "fontSize": 20,
        },
        undefined,
      ]
    }
  >
     
     
     
  </Text>
</View>
`;

exports[`<AppScreen /> renders unchanged 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`<AppText /> renders unchanged 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Avenir-Roman",
        "fontSize": 20,
      },
      undefined,
    ]
  }
>
   
</Text>
`;

exports[`<AppTextInput /> renders unchanged 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e0e0eb",
      "borderRadius": 25,
      "flexDirection": "row",
      "marginVertical": 20,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontFamily": "Avenir-Roman",
        "fontSize": 20,
        "marginLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
